README.md

```diff
@@ -15,251 +18,331 @@ The role can be used to configure:
 
-- `up`
-- `down`
+* `up` -  the connection profile is activated
+* `down` - the connection profile is deactivated
 
-It defines whether the profile is activated (`up`) or deactivated (`down`). If
-it is unset, the profile's runtime state will not be changed.
+#### `state: up`
+- For `NetworkManager`, this corresponds to `nmcli connection id {{name}} up`.
 
-The `persistent_state` setting is either `present` (default) or `absent`. If
-the `persistent_state` setting is `present` and the connection profile contains
-a `type` setting, the profile will be created or updated. If the profile is
-incomplete (lacks the `type` setting) and `persistent_state` is `present`,
-the behavior is undefined. The value `absent` makes the role ensure
-that the profile is not present on the target host.
+- For `initscripts`, this corresponds to `ifup {{name}}`.
 
+When the `state` option is set to `up`, you can also specify the `wait` option (optional):
 
-#### Example
+* `wait: 0` - initiates only the activation, but not wait until the device is fully connected.
+The connection is completed, for example after a DHCP lease received.
+* `wait: <seconds>` is a timeout that enables you to decide how long you give the device to
+activate. The default is using a suitable timeout. Note that the `wait` option is
+only supported by NetworkManager.
 
-```yaml
-network_connections:
-  - name: eth0
-    persistent_state: absent
-```
+**TODO**: `wait` different from a `zero` value is not yet implemented.
 
-Above example ensures the absence of a connection profile. If a profile with `name` `eth0`
-exists, it will be deleted.
+Note that `state: up` always re-activates the profile and possibly changes the
+networking configuration, even if the profile was already active before. As
+a consequence, `state: up` always changes the system.
 
-* For NetworkManager this deletes all connection profiles with the matching `connection.id`.
-  Deleting a profile usually does not change the current networking configuration, unless
-  the profile was currently activated on a device. In that case deleting the currently
-  active connection profile disconnects the device. That makes the device eligible
-  to autoconnect another connection (see also [rh#1401515](https://bugzilla.redhat.com/show_bug.cgi?id=1401515)).
+#### `state: down`
 
-* For initscripts it results in the deletion of the ifcfg file. Usually that
-  has no side-effect, unless some component is watching the sysconfig directory.
+- For `NetworkManager`,  it corresponds to `nmcli connection id {{name}} down`.
 
-#### Example
+- For `initscripts`, it corresponds to call `ifdown {{name}}`.
 
-```yaml
-network_connections:
-  - name: eth0
-    #persistent_state: present  # default
-    type: ethernet
-    autoconnect: yes
-    mac: 00:00:5e:00:53:5d
-    ip:
-      dhcp4: yes
-```
+You can deactivate a connection profile, even if is currently not active. As a consequence, `state: down` always changes the system.
```

Comment:
@tyll for 'nm' it seems the role should just silently accept if the profile is currently not active (or even not exists), no? Seems in code, that is currently not happening -- as described in the README here.

Reply:
It works for me to ignore state:down if the profile does not exist. The last discussion we had about this lead to make this a configuration setting: https://github.com/linux-system-roles/network/issues/70

Reply:
Yes, this covers the case where the profile does not exist. There is also the case, where the profile is already down. For `nm` provider, calling `state: down` on such a profile leaves the system unchanged. That is partly also the case for `initscripts` provider, except that the role does only very badly understand whether the profile is already down or not (https://github.com/linux-system-roles/network/blob/master/library/network_connections.py#L581).

Anyway, the sentence "always changes the system" is not correct.